test/clientconn_state_transition_test.go

```diff
@@ -164,7 +164,7 @@ func testStateTransitionSingleAddress(t *testing.T, want []connectivity.State, s
 		connMu.Unlock()
 	}()
 
-	client, err := grpc.Dial("",
+	client, err := grpc.NewClient("",
```

Comment:
This test depends on the passthrough resolver forwarding the empty string to the  dialer used below in `grpc.WithDialer(pl.Dialer())`.  The dialer doesn't use the target address provided by grpc, it simply connects to the pipe listener created at line 155. 
To maintain the same behaviour, use the passthrough scheme here.
```suggestion
client, err := grpc.NewClient("passthrough:///",
```